### PR TITLE
Issue#800 Added DmnResponseHandler interface for custom handling after decision table execution

### PR DESCRIPTION
**Pull request by AminaZoheir**

### CODE REVIEW

**Comment by tijsrademakers** [comment]
Hi,

The BPMN Engine is not depending on Spring, so we can't use @Autowired there. I understand the idea with the interface, but do you want to have one general DmnResponseHandler implementation for all DMN Engine results? I think it would make sense to allow it to be configurable on the DMN decision task in the Flowable Modeler.

**Comment by AminaZoheir** [comment]
Hello @tijsrademakers ,
I added the response handler as a custom class specified in the UI as an extension element.

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/DmnActivityBehavior.java [hunk 1]
```diff
@@ -133,17 +136,39 @@ public void execute(DelegateExecution execution) {
                 }
             }
         }
-
+        //Call custom decision table response handler if present
+        FieldExtension handlerFieldExtension = DelegateHelper.getFlowElementField(execution, EXPRESSION_DECISION_TABLE_RESPONCE_HANDLER);
+        boolean handlerFound = false;
+        if(handlerFieldExtension != null) {
```
[the remark this answers is not in the document]
Comment (filiphr) [reply modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/DmnActivityBehavior.java]: @lamtrhieu the formatting is not a big problem, we can always format it when merging it upstream. We will look soon review the PR

**Comment by tijsrademakers** [comment]
Closing this PR due to lack of activity. Sorry for the lack of communication on our end on this PR. Please reopen the PR if you still would like this functionality to be added and we will prioritize it.

**Pull request closed without merging**